### PR TITLE
feat: added optional middleware param

### PR DESCRIPTION
**Pull request by christyjacob4**

### CODE REVIEW

File: src/Route.php [hunk 1]
```diff
@@ -177,6 +184,18 @@ public function param($key, $default, $validator, $description = '', $optional =
         return $this;
     }
 
+    /**
+     * Set middleware status
+     *
+     * @return bool
+     */
+    public function middleware($useMiddleware = true): self
```
Comment (eldadfux) [inline src/Route.php:192]: Be consistent with other methods here, use just `$this->middleware`

File: src/Route.php [hunk 2]
```diff
@@ -358,4 +377,14 @@ public function getOrder(): int
     {
         return $this->order;
     }
+
+    /**
+     * Get middleware status
+     *
+     * @return bool
+     */
+    public function useMiddleWare(): bool
```
Comment (eldadfux) [inline src/Route.php:386]: Be consistent with other methods: getMiddleWare

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: src/Route.php (hunk 1 above)
Comment (christyjacob4) [reply src/Route.php]: done 👍

File: src/Route.php (hunk 2 above)
Comment (christyjacob4) [reply src/Route.php]: done 👍

**Review by eldadfux — APPROVED** [review]

**Pull request merged**